Lib/ctypes/__init__.py

```diff
@@ -312,8 +313,9 @@ def SetPointerType(pointer, cls):
     _pointer_type_cache[cls] = pointer
     del _pointer_type_cache[id(pointer)]
 
-# XXX Deprecated
 def ARRAY(typ, len):
+    import warnings
+    warnings._deprecated("ctypes.ARRAY", remove=(3, 15))
     return typ * len
```

Comment:
Do you think there's a high expected maintenance overhead or security risk here?

Per PEP 387:

> If the expected maintenance overhead and security risk of the deprecated behavior is small (e.g. an old function is reimplemented in terms of a new, more general one), it can stay indefinitely (or until the situation changes).

Reply:
it's not a big maintenance burden nor a securty issue, why do you ask?

Reply:
I ask to see if there's a special reason for `remove=(3, 15)`, or if it can be relaxed.

Reply:
What do you propose? To postpone the removal to Python 3.16?

On PyPI top 8,000 projects (at 2024-03-16), I found only 2 projects using ctypes.ARRAY. I search for regex `ctypes.ARRAY|\<ARRAY\>` (and excluded unrelated lines).

* astroid (3.1.0)
* pwntools (4.12.0)

It's about two lines:

```
astroid-3.1.0.tar.gz: astroid-3.1.0/tests/brain/test_ctypes.py: ctypes.ARRAY(3, 2)
pwntools-4.12.0.tar.gz: pwntools-4.12.0/pwnlib/adb/bootloader.py: imgarray = ctypes.ARRAY(img_info, self.num_images)
```

Before the function was only deprecated with a comment:

```py
# XXX Deprecated
def ARRAY(typ, len):
    return typ * len
```

Reply:
I'll propose to follow the part of PEP 387 I quoted above. Alternatively, since this does break known projects, to soft-deprecate instead.

But I plan to propose that in the beta period, right now I'm only gathering info. Thanks for the answer!